cadence/contracts/services/AxelarGovernanceService.cdc

```diff
@@ -0,0 +1,296 @@
+            var appCapability: &AxelarGovernanceService.Updater? = AxelarGovernanceService.account.borrow<&AxelarGovernanceService.Updater>(from: capabilityPath)
+            if appCapability == nil {
+                appCapability = AxelarGovernanceService.claimAuthCapability(provider: self.target)
+            }
+            if appCapability == nil{
+                panic("Cannot retrieve app executable capability")
+            }
+            appCapability?.update(code: self.proposedUpdate.code.decodeHex(), contractName: self.proposedUpdate.name)
+            self.executed = true
+        }
+
+    }
+
+    //Updater Resource
+    access(all) resource Updater{
+        access(self) let address: Address
+        access(self) let authCapability: Capability<&AuthAccount>
+
+        init(
+            authCapability: Capability<&AuthAccount>,
+        ) {
+            // Validate given Capability
+            if !authCapability.check() {
+                panic("Account capability is invalid for account: ".concat(authCapability.address.toString()))
+            }
+            self.address = authCapability.borrow()!.address
+            self.authCapability = authCapability
+
+        }
+
+        access(contract) fun update(code: [UInt8], contractName: String): Bool {
+            let account = self.authCapability.borrow()
+            if let account = self.authCapability.borrow() {
+                account.contracts.update__experimental(name: contractName, code: code)
+            }
+            return true
+        }
+    }
+
+
+
+    /**********\
+    |* Events *|
+    \**********/
+
+    pub event ProposalScheduled(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        eta: UInt64
+    )
+
+    pub event ProposalCancelled(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        eta: UInt64
+    )
+
+    pub event ProposalExecuted(
+        proposalHash: String,
+        target: Address,
+        proposedCode: String,
+        timestamp: UInt64
+    )
+
+
+    init(gateway: Address, governanceChain: String, governanceAddress: String, minimumTimeDelay: UInt64){
+        if (governanceChain.length == 0 || governanceAddress.length == 0){
+            panic("InvalidAddress")
+        }
+
+
+        self.gateway = gateway
+        self.governanceChain = governanceChain
+        self.governanceAddress = governanceAddress
+        self.minimumTimeDelay = minimumTimeDelay
+        self.SELECTOR_SCHEDULE_PROPOSAL = Crypto.hash("scheduleProposal".utf8, algorithm: HashAlgorithm.KECCAK_256)
+        self.SELECTOR_CANCEL_PROPOSAL = Crypto.hash("cancelProposal".utf8, algorithm: HashAlgorithm.KECCAK_256)
+        self.inboxAccountCapabilityNamePrefix = "GovernanceUpdaterInbox_"
+        self.prefixAuthCapabilityName = "GovernanceUpdaterCapability_"
```

Comment:
Looks like this is used as a prefix to the Updater resource in storage. Recommend renaming to reflect the updated use - e.g. `updaterStoragePathPrefix` or similar - since there are no Updater Capabilities issued or dealt with

Reply:
Fixed!